### PR TITLE
[WIP] Feature/3651 add flake8

### LINKED ISSUE #3651: Automate PEP8 linting tests
We should automate Python linting in the API and CMS like we do for JavaScript in the CMS.

See openFEC PR https://github.com/fecgov/openFEC/pull/4259 

- [ ] Add linting to pytests

### PR DESCRIPTION
**Pull request by lbeaufort**
- [ ] TODO: Test with manual deploy?

## Summary (required)

- Resolves #3651 
- Automate pep8 checking with `pytest-flake8` (See "how to test" for instructions for use)

max line length = 120 chars

**Skipping**
- E128 continuation line under-indented for visual indent
- E127 continuation line over-indented for visual indent
- W503 line break before binary operator

## How to test the changes locally

- check out this branch 
- `pip install -r requirements.txt`
- `pytest` includes `pytest-flake8` PEP8 linting
- `pytest --no-linting` skips `pytest-flake8` PEP8 linting
- `pytest --linting` only runs `pytest-flake8` PEP8 linting
- `pytest --cache-clear --linting` to run without cached results https://github.com/tholo/pytest-flake8/issues/62
- You can still run one tests with `pytest tests/test_elections.py --no-linting` or `--linting` or don't specify either

## Related PR
OpenFEC linting https://github.com/fecgov/openFEC/pull/4259

### CODE REVIEW

File: fec/search/management/commands/scrape_cms_pages.py [hunk 1]
```diff
@@ -2,61 +2,52 @@
...
+
 class Command(BaseCommand):
     help = 'Scrapes pages from the CMS into JSON for indexing on DigitalGov Search'
 
     def add_arguments(self, parser):
         parser.add_argument(
             '--descendants_of',
             type=str,
-            help='Path of the parent page whose descendents to get, ex. /home/legal-resources/'
+            help='Path of the parent page whose descendents to get, ex. /home/legal-resources/',
```
Comment (rfultz) [inline fec/search/management/commands/scrape_cms_pages.py:21]: @lbeaufort any idea why we're using the trailing comma here and lines 27 and 38 but not line 31?

File: fec/search/management/commands/scrape_web_app_pages.py [hunk 2]
```diff
@@ -36,8 +37,11 @@ def handle(self, *args, **options):
                 "created": "2017-04-01",
                 "language": "en",
                 "promote": "true",
-                "description": "See how candidates and committees raise and spend money in federal elections. This financial data helps voters make informed decisions.",
-                "tags": "data, finance data, contributions, candidates, committees, pacs, super pacs, disclosure",
+                "description": "See how candidates and committees raise and spend \
```
Comment (rfultz) [inline fec/search/management/commands/scrape_web_app_pages.py:40]: I'd like to see how these render in the source—are we escaping only the line break or all of the whitespace after the \ ?

File: fec/search/utils/search_indexing.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def handle_page_edit_or_create(page, method):
         # Make sure the page is live and public
         try:
             queried_page = Page.objects.live().public().get(id=page.id)
-        except:
+        except Exception:
```
Comment (rfultz) [inline fec/search/utils/search_indexing.py:134]: I've seen this a couple times. ?

File: fec/search/management/commands/scrape_cms_pages.py (hunk 1 above)
Comment (lbeaufort) [reply fec/search/management/commands/scrape_cms_pages.py:21]: @rfultz I used `Black`, which is a code auto=formatter (https://github.com/psf/black) - I'm guessing it's the fact that it's on multiple lines? I can clean these up to be consistent.

File: fec/home/templatetags/filters.py [hunk 4]
```diff
@@ -150,20 +157,18 @@ def get_social_image_path(identifier):
     # TODO: combine with fec/data/templatetags/filters.py ?
     # Called by meta-tags.html
     # """
-    imageFilename = identifier
+    image_filename = identifier
```
[the remark this answers is not in the document]
Comment (lbeaufort) [reply fec/home/templatetags/filters.py:160]: The return value of this function should be the same, regardless of how the variables are named. I just updated these to be consistent with Python-style `snake_case`.

**Comment by lbeaufort** [comment]
De-prioritizing  #3651, will re-open and rebase

**Pull request closed without merging**